torch_geometric/transforms/add_metapaths.py

```diff
@@ -168,3 +180,16 @@ def sample_adj(self, adj: SparseTensor) -> SparseTensor:
         prob = (self.max_sample * (1. / deg))[row]
         mask = torch.rand_like(prob) < prob
         return adj.masked_select_nnz(mask, layout='coo')
+
+    def _get_edge_attr(self, data: HeteroData,
+                       edge_type: EdgeType) -> torch.Tensor:
+        if self.weighted:
+            if self.use_edge_attr_as_weights and \
+                    edge_type in data.edge_attr_dict.keys():
+                edge_attr = data[edge_type].edge_attr
```

Comment:
I'm thinking if we could have users to have a custom edge attribute name that can be used to specify weights.
E.g. the user might want to call it `edge_weights` or `edge_multiplicity` or whatever, so that `edge_attr` can still be used for other purposes (such as in message function of the `MessagePassing` class)`.

For example, [here](https://github.com/pyg-team/pytorch_geometric/blob/575611f4f5e2209c7923dba977a1eebc207bd2e2/torch_geometric/loader/neighbor_loader.py#L97), `time_attr` is defined as a string (we could name it as `time_col` or other names.

Reply:
Good idea. I think I can get rid of the second extra argument and just assume that edge_multiplicity is defined then it should be used.